### PR TITLE
CPLAT-14012: Migrate to null-safety

### LINKED ISSUE #37: Support null-safety
Looks like the only non dev-dependency that doesn't support null safety yet is `package:io`, which now has an open [PR](https://github.com/dart-lang/io/pull/67) for null safety.

### PR DESCRIPTION
**Pull request by evanweible-wf**
# [CPLAT-14012](https://jira.atl.workiva.net/browse/CPLAT-14012)
Issue Status

- NNBD migration
  - Only notable change to the public API is that `AstVisitingSuggestor.context` (mixin used to write AST-based suggestors) will throw if `context` is accessed when it shouldn't be. Previously it just would have been null.
  - Use the builder-approach for `Mock`s: see https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md
- Upgrade to latest `package:analyzer` and address some deprecated API usages
- Unskip a test that no longer needs to be skipped
- Added a readme to `example/` for the pub site (the package score was being docked because it was missing)
- No longer run CI on Dart 2.7.2 since it now requires 2.12.0
- Will be released as v1.0.0

Closes #37

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/file_context.dart [hunk 1]
```diff
@@ -37,29 +37,34 @@ class FileContext {
   /// text, which is useful for the creation of [SourcePatch]es.
   SourceFile get sourceFile =>
       _sourceFile ??= SourceFile.fromString(sourceText, url: Uri.file(path));
-  SourceFile _sourceFile;
+  SourceFile? _sourceFile;
```
Comment (greglittlefield-wf) [inline lib/src/file_context.dart:40]: #nit This private variable and getter could be replaced with a `late` variable

https://dart.dev/null-safety/understanding-null-safety#lazy-initialization
```dart
final late SourceFile sourceFile = SourceFile.fromString(sourceText, url: Uri.file(path);
```

Same with `sourceText`/`_contents`

**Review by greglittlefield-wf — APPROVED** [review]

File: lib/src/file_context.dart (hunk 1 above)
Comment (evanweible-wf) [reply lib/src/file_context.dart:40]: Woah! I had no idea you could do this, that's awesome!

**Review by greglittlefield-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passes
- [x] pub publish --dry-run passes and next release will opt the package into null-safety

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**